src/App.tsx

```diff
@@ -14,11 +15,103 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[];
+};
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: ['Jam'],
+  };
+
+  addSelectedGood = (good: string) => {
+    const { selectedGood } = this.state;
+
+    this.setState({ selectedGood: [...selectedGood, good] });
+  };
+
+  removeSelectedGood = (good: string) => {
+    const { selectedGood } = this.state;
+
+    this.setState({ selectedGood: selectedGood.filter(item => item !== good) });
+  };
+
+  clearSelectedGood = () => {
+    this.setState({ selectedGood: [] });
+  };
+
+  showListGoods = () => {
+    const { selectedGood } = this.state;
+
+    switch (selectedGood.length) {
+      case 0:
+        return 'No goods selected';
+      case 1:
+        return (
+          selectedGood.toString().endsWith('s')
+            ? `${selectedGood} are selected`
+            : `${selectedGood} is selected`
+        );
+
+      default:
+        return (
+          `${selectedGood.slice(0, -1)
+            .join(', ')}
+            and
+            ${selectedGood[selectedGood.length - 1]}
+            are selected`
+        );
+    }
+  };
+
+  render() {
+    const { selectedGood } = this.state;
+
+    return (
+      <div className="App">
+        <div className="App__header">
+          {selectedGood.length > 0
+            && (
+              <button
+                type="button"
+                className="App__btn App__btn--clear"
+                onClick={() => this.clearSelectedGood()}
```

Comment:
if you don't pass any arguments, no need for arrow function

```suggestion
                onClick={this.clearSelectedGood}
```